lib/client/ClientRenderingWorkflow.js

```diff
@@ -28,13 +28,17 @@ var ClientRenderingWorkflow = {
             var errorViewMappingForRoute = router.errorViewMapping;
             var onRouteStart = router.onRouteStart;
             var onRouteComplete = router.onRouteComplete;
-            var routeArguments = Array.prototype.slice.call(arguments, 0);
 
             var isFirstPageLoad = !currentLayout;
             var layoutForRouteIsSameAsCurrent = isFirstPageLoad || currentLayout.isSameTypeAs(LayoutForRoute);
             var shouldUseCurrentLayout = !isFirstPageLoad && layoutForRouteIsSameAsCurrent;
             var layout = shouldUseCurrentLayout ? currentLayout : new LayoutForRoute();
             var layoutDelegate = LayoutDelegate.from(layout, LayoutForRoute);
+            var routeArguments = new Array(arguments.length);
+
+            for (var i = 0; i < routeArguments.length; i++) {
+                routeArguments[i] = arguments[i];
```

Comment:
yup, was trying to be like @petkaantonov - https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#32-leaking-arguments